### DISCUSSION TITLE
multi-tenancy indexing

### DISCUSSION THREAD

**Discussion by oronsh** [lnx-search/lnx#36] [Q&A] [answered]
Hello,
Thank you for this amazing project!
I've noticed that `tantivy` uses a thread pool per index and so is the `IndexWorker` in `lnx`?
I would like to know how this will behave when a lot of indexes are being written to at the same time.
Lets say I have 10,000 indexes opened at the same time, will it open NUM_THREADS * 10,000 threads?
Thanks a lot!

**Comment by ChillFish8** [comment, marked as answer]
This is mentioned in the docs https://lnx.rs/#tag/Managing-indexes which gives a bit more info on this, but yes this will open 10,000 * (calculated number of threads) which is generally something I wouldn't advise doing. 

I do plan on working on some sort of shared / dynamic size executor but for the most part, It's just really not recommended to make lots of small indexes as it is unbelievably inefficient to do so in lnx, you're better of making one index and merging all the data together as best you can, at least in lnx's case.

If you *need* a lot of very small indexes something like [sonic](https://github.com/valeriansaliou/sonic) might be better suited as it's a lot more optimised for small resources footprint vs performance on large indexes. 

Hopefully, this clears up any questions you might have.

**Reply by oronsh** [reply]
Yes, this clears things out for me, I'm building an indexing server that should support multi-tenancy and checked out the architecture to learn so this is why I've asked about this matter.
Thank you so much for your thorough answer and fast reply!
